src/app/campaign/inventory/goal-form.component.ts

```diff
@@ -98,24 +103,29 @@ export class GoalFormComponent implements OnInit {
 
   ngOnInit() {
     this.goalForm = this.formController.control as FormGroup;
+    this.velocitySub = this.goalForm.get('velocity').valueChanges.subscribe(velocity => this.setDailyMinimum(velocity));
+    this.goalMinSub = combineLatest([this.goalForm.get('totalGoal').valueChanges, this.goalForm.get('dailyMinimum').valueChanges])
+      .pipe(distinctUntilChanged())
+      .subscribe(([goal, min]) => this.setVelocity(goal, min));
   }
 
-  getVelocity(): string {
-    const min = this.dailyMinimum;
-    const goal = this.totalGoal;
-    if (min && min >= goal) {
-      return 'fastly';
-    } else if (min) {
-      return '';
-    } else {
-      return 'evenly';
+  ngOnDestroy() {
+    if (this.velocitySub) {
+      this.velocitySub.unsubscribe();
     }
+    if (this.goalMinSub) {
+      this.goalMinSub.unsubscribe();
+    }
+  }
+
+  setVelocity(totalGoal: number, dailyMinimum: number) {
```

Comment:
don't emit velocity changes set from the totalGoal and dailyMin. a) prevents circular update cycle b) velocity is really just UI state, so it doesn't need to push those updates out anywhere else